### PR TITLE
feat(gatsby-plugin-google-analytics): enable more options.

### PR DESCRIPTION
**Pull request by tinder-kyleboss**
### More options for gatsby-plugin-google-analytics

🥁The *new* options are🥁
- `allowAdFeature`
- `dataSource`
- `queueTime`
- `forceSSL`
- `transport`

### Implementation details
This included breaking out `knownOptions` into ~three~ two factions:
- createOnly
- general
~- aLaCarte~

❤️ Complete with tests ❤️

## Related Issues

#15168

### CODE REVIEW

**Comment by tinder-kyleboss** [comment]
Would it be possible for someone that has the privileges to re-run the `e2e_tests_production_runtime`? This is my first contribution so I am not too sure if the end-to-end tests are flakey. The failures do not *seem* related to the changes made in this PR :) (obvi not ruling that out though!)

**Comment by tinder-kyleboss** [comment]
Looks like it passed :) Thanks fam. 👍

File: packages/gatsby-plugin-google-analytics/src/gatsby-ssr.js [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+    allowLinker: `boolean`,
+  },
+  general: {
+    allowAdFeatures: `boolean`,
+    dataSource: `string`,
+    queueTime: `number`,
+    forceSSL: `boolean`,
+    transport: `string`,
+  },
+  aLaCarte: {
```
Comment (wardpeet) [inline packages/gatsby-plugin-google-analytics/src/gatsby-ssr.js]: could you elaborate on why this is added? I can't seem to find this being used in your changes?

Comment (tinder-kyleboss) [reply packages/gatsby-plugin-google-analytics/src/gatsby-ssr.js]: Absolutely! Good question 👍 

The `aLaCarte` section is for options that are not directly from the Google Analytics API/docs, but rather options specific to this Gatsby Plugin. 
The context is that the `anonymize` option is not a Google Analytics option (not to be confused with the very similar GA-official option `anonymizeIp`).

Upon further contemplation, the name is kind of vague. Thus I will be renaming the section to `pluginSpecific` and adding a comment to clarify what the section is for. 🔥

**Comment by tinder-kyleboss** [comment]
Quick update @wardpeet!

I decided too get rid of the `aLaCarte` section of the `knownOptions` object. Looking at it again, I realized it has no functional use being there. Thanks for catching this one. ⚾️ 

Feel free to let me know if you see anything else!

**Comment by tinder-kyleboss** [comment]
I am not really sure what the ENOAUDIT message is for. Is this something I broke and can fix on my end?

**Comment by wardpeet** [comment]
Successfully published:
 - gatsby-plugin-google-analytics@2.1.5

**Comment by flegall** [comment]
Hi, I don't know if it's relevant to this PR but after bumping the google analytics plugin last night, the google analytics ssr output is broken : 
Screenshot 2019-07-30 at 14 35 36

It didn't close the `if () {` bracket. 
It doesn't impact the navigation and doesn't occur when running in develop mode.

You can reproduce it on this branch : https://github.com/flegall/florent-legall-dot-com/tree/google_analytics_ssr_issue : 
`yarn && yarn start:prod`

**Comment by tinder-kyleboss** [comment]
Darn! Thank you so much for catching this @flegall. 🙏 

I just opened up the PR #16223 to fix this issue.

**Comment by flegall** [comment]
Nice ! Most of the reporting work was actually done by renovate & cypress !

I'm receiving PRs from renovate (that get merged automatically if they pass).
Latest PR to upgrade gatsby failed due to an error in the few cypress e2e tests I've written for my blog. 
Cypress failed because it caught the error in the JS code caused by the missing "}" !

It could have gone unnoticed though, as the whole site was working.

**Pull request merged**